### PR TITLE
Defaulting None for optional fields

### LINKED ISSUE #70: Always require optional field in python target
```
record foo (
  text? optional-data
);
```

Above `foo` record compiled to

```
class Foo:
    ...

    def __init__(self, optional_data: typing.Optional[str]) -> None:
        ...
```

It requires `optional_data` as a argument to initiate class `Foo`, although `optional_data` is optional.

### PR DESCRIPTION
**Pull request by earlbread**
This fixes #70.

### CODE REVIEW

File: test/nirum_fixture/fixture/foo.nrm [hunk 1]
```diff
@@ -100,3 +100,10 @@ record person (
 record people (
   {person} people
 );
+
+record product (
+    text name,
+    int64? price,
+    bool sale,
+    uri? url,
+);
```
Comment (dahlia) [inline test/nirum_fixture/fixture/foo.nrm]: We should test union tags as well e.g.:

~~~~ nirum
union union-test = tag-a
                 | tag-b (text a, text? b, int64 c, int64 d?);
~~~~

File: test/python/primitive_test.py [hunk 2]
```diff
@@ -249,3 +249,9 @@ def test_service():
         PingService().ping(nonce=u'nonce')
     with raises(TypeError):
         PingService().ping(wrongkwd=u'a')
+
+
+def test_optional_initializer_test():
+    product = Product(name=u'coffee', sale=False)
+    assert product.price is None
+    assert product.url is None
```
Comment (dahlia) [inline test/python/primitive_test.py]: We should test union tags as well e.g.:

~~~~ python
def test_union_tags_optional_initializer():
    t = TagB(a=u'test', c=123)
    assert t.a == u'test'
    assert t.b is None
    assert t.c == 123
    assert t.d is None
~~~~

File: test/python/primitive_test.py [hunk 3]
```diff
@@ -249,3 +249,9 @@ def test_service():
         PingService().ping(nonce=u'nonce')
     with raises(TypeError):
         PingService().ping(wrongkwd=u'a')
+
+
+def test_optional_initializer_test():
```
Comment (dahlia) [inline test/python/primitive_test.py]: Since the function name already starts with `test_` we need no `_test` suffix here.

**Comment by earlbread** [comment]
@dahlia I fixed things that you mentioned. Thank you!

**Pull request merged**